fsm/committee.go

```diff
@@ -34,20 +34,40 @@ func (s *StateMachine) FundCommitteeRewardPools() lib.ErrorI {
 	subsidizedCount := uint64(len(subsidizedChainIds))
 	// if there are no subsidized committees or no mint amount
 	if subsidizedCount == 0 || totalMintAmount == 0 {
-		return nil
+		err = &lib.Error{Msg: "Nothing to do"}
+		return
 	}
 	// calculate the amount left for the committees after the parameterized DAO cut
 	mintAmountAfterDAOCut := lib.Uint64ReducePercentage(totalMintAmount, govParams.DaoRewardPercentage)
 	// calculate the DAO cut
-	daoCut := totalMintAmount - mintAmountAfterDAOCut
+	daoCut = totalMintAmount - mintAmountAfterDAOCut
+
+	// calculate the amount given to each qualifying committee
+	// mintAmountPerCommittee may truncate, but that's expected,
+	// less mint will be created and effectively 'burned'
+	mintAmountPerCommittee = mintAmountAfterDAOCut / subsidizedCount
+
+	return daoCut, totalMintAmount, mintAmountPerCommittee, nil
+}
+
+// FundCommitteeRewardPools() mints newly created tokens to protocol subsidized committees
+func (s *StateMachine) FundCommitteeRewardPools() lib.ErrorI {
+	daoCut, _, mintAmountPerCommittee, err := s.GetBlockMintStats(s.Config.ChainId)
+	if err != nil {
+		if err.Error() == "Nothing to do" {
```

Comment:
same here, when changing to a pre-existing error from `fsm/errors.go` please do compare it against the error using `errors.Is`